lib/spack/spack/cmd/__init__.py

```diff
@@ -208,13 +213,56 @@ def disambiguate_spec(spec, env, local=False, installed=True, first=False):
         installed (boolean or any, or spack.database.InstallStatus or iterable
             of spack.database.InstallStatus): install status argument passed to
             database query. See ``spack.database.Database._query`` for details.
+        first (boolean, default False): in case of multiple matches, simply
+            return the first spec in the list.
+        compatible (boolean, default False): attempt to locate a package that
+            does not conflict with any of the currently loaded packages.
+        preferred (boolean, default False): use information gleaned from
+            packages.yaml and the current platform to assist in narrowing
+            down to specific package. (currently uses versions, variants,
+            architechture and compiler). If a preference eliminates all
+            results, the preference will not be honored.
     """
     hashes = env.all_hashes() if env else None
-    return disambiguate_spec_from_hashes(spec, hashes, local, installed, first)
+    return disambiguate_spec_from_hashes(spec, hashes, local, installed, first,
+                                         compatible, preferred)
+
+
+def _find_already_loaded():
+    spechash = {}
+    loadedhashes = os.environ.get(
+        spack.user_environment.spack_loaded_hashes_var, '').split(':')
+    matching_specs = spack.store.db.query(hashes=loadedhashes)
+    for spec in matching_specs:
+        for node in spec.traverse():
```

Comment:
This is not an accurate check. A spec can be loaded without loading its dependencies.